docs/todo-due-dates.md

```diff
@@ -0,0 +1,134 @@
+- Feature Name: `due_dates`
+- Start Date: `2020-11-09`
+- RFC PR: (leave this empty)
+- Checkup Issue: (leave this empty)
+
+# Summary
+
+[summary]: #summary
+
+This RFC proposes enhancing the todo functionality with due days, allowing authors to add accountability and commitment to fix the todo items by a specified amount of time.
+
+# Motivation
+
+[motivation]: #motivation
+
+The lint todo functionality is intended to seamlessly integrate with existing linting tools, allowing the incremental fixing of linting errors on very large projects. However, one undesirable side effect of this is that developers may forget about fixing the todo items leading to the accumulation of technical debts.
+
+By adding a due date to the todo items, developers will now have a commitment to fix the linting problems because todo items that are past-due will once again be considered errors.
+
+# Pedagogy
+
+[pedagogy]: #pedagogy
+
+- There will be documentation on how to configure when todo items are due to be fixed.
+- There will be a common config and CLI argument to specify the number of days todo items are due to be fixed.
+- There will be a common config and CLI argument to specify the number of days an item becomes an error after its due date.
+
+# Details
+
+[details]: #details
+
+## Requirements
+
+- By default, todos are added with no due dates.
+- Authors can specify the number of days the todos must be fixed after its creation date.
+- Authors can specify the number of days the todos are flagged as `error`s after they are past-due. The default value is `0`.
```

Comment:
Why don't we instead allow for the configuration of both a warning threshold and an error threshold, individually. This would be the logic:

- if no warn or error thresholds are given, no due dates are configured
- if a warn threshold is given, in days, a todo transitions to a warn following that number of days past its creation date
- if an error threshold is given, in days, a todo transitions to an error following that number of days past its creation date
- if both a warn threshold _and_ and error threshold is given, a todo transitions first to a warn following that number of days past its creation date, then to an error following that number of days past the warn (warn + error days past creation date)

Reply:
I have two concerns with this:

- error threshold will have a different behavior if a warn threshold is also passed which is confusing. I believe it is better if an argument always has the same outcome. For example, error threshold is always the amount of time past its creation date.
- todos come from errors not warnings, so todos should not be in a state where they are marked as warnings permanently.

This is why I proposed `daysToFix`, where by default it is when todos become errors. If authors want a transitional period they can increase the additional number of days until it becomes an error.

Thoughts?

Reply:
> error threshold will have a different behavior if a warn threshold is also passed which is confusing. I believe it is better if an argument always has the same outcome. For example, error threshold is always the amount of time past its creation date.

That's fine. I can understand this. We can make errors be an absolute value from creation date too, but it would need to be a number greater than warn if both were given.

> todos come from errors not warnings, so todos should not be in a state where they are marked as warnings permanently.

That's a valid point. What if we keep `daysToError` mandatory, but optionally allow `daysToWarn` to be provided, however you cannot use `daysToWarn` on its own. What I'm trying to do is keep the language consistent and inline with the severity terminology. Fix has a very specific meaning in linting, and it's not generally related to severity, which the todo feature _is_.

Reply:
Yes, I like that. I'll adjust.